src/main/scala/org/constellation/p2p/PeerAPI.scala

```diff
@@ -158,20 +158,23 @@ class PeerAPI(override val ipManager: IPManager[IO])(
         path("faucet") {
           entity(as[SendToAddress]) { sendRequest =>
             // TODO: Add limiting
+            // TODO: Chain
             if (sendRequest.amountActual < (dao.processingConfig.maxFaucetSize * Schema.NormalizationFactor) &&
                 dao.addressService
                   .lookup(dao.selfAddressStr)
                   .unsafeRunSync()
                   .map { _.balance }
                   .getOrElse(0L) > (dao.processingConfig.maxFaucetSize * Schema.NormalizationFactor * 5)) {
 
-              val tx = createTransaction(
-                dao.selfAddressStr,
-                sendRequest.dst,
-                sendRequest.amountActual,
-                dao.keyPair,
-                normalized = false
-              )
+              val tx = dao.transactionService
+                .createTransaction(
+                  dao.selfAddressStr,
+                  sendRequest.dst,
+                  sendRequest.amountActual,
+                  dao.keyPair,
+                  normalized = false
+                )
+                .unsafeRunSync()
```

Comment:
```APIDirective.handle(addressLookup >> tx)){
other.unsafeRunAsyncAndForget
complete(_)
}```

Reply:
I've added TODO statement above to chain it into one operation